### PR TITLE
refact: Directives as single tokens

### PR DESCRIPTION
**Pull request by keyvank**

### CODE REVIEW

**Comment by keyvank** [comment]
This is mandatory since we are not lexing endlines coming after each directive, i.e we can't recognize where does a directive end, e.g:

`#define A 2 * 3`

is currently being interpreted as `#define A 2` and then comes a `* 3`

**Review by Glyphack — APPROVED** [review]

File: tests/output/inp_preprocess.c_asm_output.txt [hunk 1]
```diff
@@ -19,6 +20,18 @@ ret
 mov rsp, rbp
 pop rbp
 ret
+global function
+function:
```
Comment (amiremohamadi) [inline tests/output/inp_preprocess.c_asm_output.txt]: seems it caused a problem with the `ifdef` keyword. `function` should not be defined twice ([L12](https://github.com/keyvank/30cc/blob/9935bd4987cf59076ecad2080919c47a52c684cc/tests/output/inp_preprocess.c_asm_output.txt#L12), [L24](https://github.com/keyvank/30cc/blob/9935bd4987cf59076ecad2080919c47a52c684cc/tests/output/inp_preprocess.c_asm_output.txt#L24.)).

Comment (keyvank) [reply tests/output/inp_preprocess.c_asm_output.txt]: yeah, lemme check

**Comment by keyvank** [comment]
@amiremohamadi Fixed. Could you recheck?

**Comment by amiremohamadi** [comment]
LGTM! :tada:

**Pull request merged**